src/com/cdd/bao/importer/TemplateChecker2.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * BioAssay Ontology Annotator Tools
+ * 
+ * (c) 2014-2017 Collaborative Drug Discovery Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License 2.0
+ * as published by the Free Software Foundation:
+ * 
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.cdd.bao.importer;
+
+import com.cdd.bao.template.*;
+import static com.cdd.bao.template.Schema.*;
+import com.cdd.bao.util.*;
+
+import java.io.*;
+import java.util.*;
+import java.util.function.Consumer;
+
+import org.json.*;
+
+/*
+	Loads up a schema and runs through the template content, looking for possible flaws that could cause trouble
+	further down the line.
+*/
+
+public class TemplateChecker2
+{
+	private Schema schema;
+	private Vocabulary vocab;
+	private Consumer<List<Diagnostic>> consumer;
+
+	public static final class Diagnostic
+	{
+		List<Group> groupNest; // nested group location of issue in schema tree
```

Comment:
Just make this a struct-like container class, i.e. make the members public and remove the getter/setter methods.